### PR TITLE
docs: Add user guide for Gateway API State Metrics

### LINKED ISSUE #1823: Integrate david-martin / gateway-api-state-metrics in docs
*Description*:
@david-martin has created a Grafana dashboard to help monitor Gateway API resource https://github.com/david-martin/gateway-api-state-metrics. This dashboard aggregates resource status conditions and can serve to improve Day 2 operations at scale. 

Taking a quick look at the repo, it should work out of the box with Envoy Gateway, since the metrics are based off the K8s API server.

We could add a docs page to show integration with this dashboard

### PR DESCRIPTION
**Pull request by david-martin**
**What type of PR is this?**

Docs, new user guide

**What this PR does / why we need it**:

This PR adds a new User Guide for installing and using [Gateway API State Metrics](https://github.com/Kuadrant/gateway-api-state-metrics) (metrics, dashboards and alerts).

**Which issue(s) this PR fixes**:

Fixes #1823

### CODE REVIEW

File: docs/latest/user/gateway-api-metrics.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Follow the steps from the [Quickstart Guide](quickstart.md) to install Envoy Gateway and the example manifest.
+Before proceeding, you should be able to query the example backend using HTTP.
+
+[Kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/) is used to install the metrics stack.
+Install Kustomize by following the instructions on the kustomize site.
+Once Kustomize is installed, run the following commands to install the metrics stack, with the Gateway API State Metrics configuration, on your kubernetes cluster.
+
+```shell
+kustomize build github.com/kuadrant/gateway-api-state-metrics.git/examples/kube-prometheus | docker run --rm -i ryane/kfilt -i kind=CustomResourceDefinition | kubectl apply --server-side -f -
```
Comment (Xunzhuo) [inline docs/latest/user/gateway-api-metrics.md:16]: It is possible to release the yaml in upstream, so that people can use it easier

**Review by Xunzhuo — COMMENTED** [review]
Thanks for working on it, did not test it yet, raise a quesion here.

File: docs/latest/user/gateway-api-metrics.md (hunk 1 above)
Comment (Xunzhuo) [reply docs/latest/user/gateway-api-metrics.md:16]: Yep, like https://github.com/envoyproxy/gateway/releases/download/latest/install.yaml, this is not blocking this PR, just raise a question.

Comment (Xunzhuo) [reply docs/latest/user/gateway-api-metrics.md:16]: image

Comment (david-martin) [reply docs/latest/user/gateway-api-metrics.md:16]: Good idea.
I'll use the kustomize cmds to generate yaml and commit that.
That should avoid potential errors (like in the screenshot above) and remove the dependency on kustomize for installing.

Comment (david-martin) [reply docs/latest/user/gateway-api-metrics.md]: @Xunzhuo I've updated the docs to simplify things (no kustomize or docker run needed) by referencing bundle yaml files.
This should also fix the error you were seeing.

**Review by zirain — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**